lib/srv/discovery/common/watcher.go

```diff
@@ -139,6 +159,8 @@ func (w *Watcher) Start() {
 
 // fetchAndSend fetches resources from all fetchers and sends them to the channel.
 func (w *Watcher) fetchAndSend() {
+	w.preFetchHookFn()
```

Comment:
nit: `WithPreFetchHookFn` can technically set the hook to nil. some nil/safety check would be nice.

Reply:
Added nil check 👍